lib/base.ts

```diff
@@ -36,6 +36,13 @@ export class TranspilerBase {
     }
   }
 
+  assert(c: ts.Node, condition: boolean, reason: string): void {
+    if (!condition) {
+      this.reportError(c, reason);
+      throw new Error(reason);
```

Comment:
Yeah assertions should, but in `ts2dart` it has turned out that it's super useful to continue program execution to find *all* the errors in one go (or as many as possible), so that users don't have to laboriously fix errors one by one, and then re-run the (rather slow) tool.

In particular things like Promise ctor parameter count, or callback parameter count are things that might end up being incorrect in user code for some reason or another. We should not assert there, we should reportError as much as possible.